services/service-core/src/auth/keycloak.service.ts

```diff
@@ -54,13 +54,13 @@ export class KeycloakService {
       );
 
       if (!getTokenData.ok) {
-        throw new Error('Error: Fetch error');
+        throw new Error(getTokenData.statusText);
```

Comment:
Make the error message more describing  like 
```typescript
throw new Error(`Couldn't fetch token data, status: ${getTokenData.status})
```

Reply:
Take it even further. Whenever you write an error message, mind that when you see this message in a log you won't have the surrounding context such as code, etc. Imagine you are reading app logs or see a http response that has only this piece of text "Couldn't fetch token data, status: 400". How long will it take you to realize what status is mentioned there? Where exactly this error happens? Does it give clear understanding if the problem?

- look at the code, what does it do? essentially you are trying to get an admin token for the app, not some abstract "token data". mention that it in the message
- consider broader context, does it allow to quickly understand and pinpoint the problem? add some clues or breadcrumbs for yourself or another fellow developer to ease up debugging
- remember that error messages are usually read and acted upon by people who have no idea about the code behind it. it is imperative for them to understand at least the essence of the problem without relying on their knowledge of the code
- remember that error messages may be exposed to general public. while descriptive and detailed, they must not leak any sensitive information or something that could be used to compromise the system

For this particular case I would suggest something like
`[KeycloakService.getAdminToken] Failed to fetch admin token. HTTP Status: 403.`

But you could go even further by analyzing the  response from keycloak that likely has a more descriptive error and including that message into your error message. This way it would be even more clear and descriptive.

Reply:
Consider the example I gave you in another repository (see my message in slack). There's code for custom error and subclassing from built-in Error class. This would be even better way to go.

Reply:
always remember: for every minute you save on writing good, descriptive error message other people will pay with hours of tiresome debugging